### ISSUE TITLE
Implement `model2_part2` training with constraint

### ISSUE THREAD

**Issue by wrongu** [pkawale/StitchNet#14]
Basic idea for M2:

    loss = loss_fn(...) + lambda_ * weight_regularization

where `weight_regularization` measures the (squared?) euclidean distance between `model2_part2.parameters()` frozen from after stitching and the same parameters during the second fine-tuning stage.

We then run this for many values of `lambda_` to map out the loss-vs-learning curve.
